modules/image-loading-optimization/module.php

```diff
@@ -0,0 +1,264 @@
+<?php
+namespace Elementor\Modules\ImageLoadingOptimization;
+
+use Elementor\Core\Base\Module as BaseModule;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Module extends BaseModule {
+
+	private $min_priority_img_pixels = 50000;
+
+	public function get_name() {
+		return 'image-loading-optimization';
+	}
+
+	public function __construct() {
+		parent::__construct();
+
+		add_action( 'wp_get_attachment_image_attributes', [ $this, 'remove_get_attachment_loading_attributes' ], 10, 3 );
+		add_action( 'wp_content_img_tag', [ $this, 'remove_content_img_tag_loading_attributes' ], 10, 3 );
+		add_filter( 'the_content', [ $this, 'filter_images' ], 10, 1 );
+	}
+
+	public function remove_get_attachment_loading_attributes( $attr, $attachment, $size ) {
+		unset( $attr['fetchpriority'] );
+		unset( $attr['loading'] );
+		return $attr;
+	}
+
+	public function remove_content_img_tag_loading_attributes( $filtered_image, $context, $attachment_id ) {
+		$tag = new \WP_HTML_Tag_Processor( $filtered_image );
+		if ( $tag->next_tag() ) {
+			$tag->remove_attribute( 'fetchpriority' );
+			$tag->remove_attribute( 'loading' );
+		}
+		return $tag->get_updated_html();
+	}
```

Comment:
Both of these filters remove core's attributes after they are applied, which gets the desired result, but means that unnecessary code is running to calculate these values before discarding them. Instead, I wonder if we could do something like the following to keep core from applying these attributes in the first place?

```php
add_filter( 'wp_lazy_loading_enabled', '__return_false' );
add_action( 'init', function () {
	wp_high_priority_element_flag( false );
} );
```

Reply:
Yes this make sense. I have added it  to the constructor.